test-data/samples/crawl2.py

```diff
@@ -0,0 +1,863 @@
+#!/usr/bin/env python3.4
+
+"""A simple web crawler."""
+
+# This is cloned from <asyncio>/examples/crawl.py,
+# with type annotations added (PEP 484).
+#
+# This version (crawl2.) has also been converted to use `async def` +
+# `await` (PEP 492).
```

Comment:
It' even more complex, and that's why I have this Matrix test and the misc/asyncio_matrix.py program. Turns out whether an async def is acceptable to yield from doesn't depend on how the async def is decorated, but it does depend on how the generator containing the yield from is decorated. Go figure! I haven't figured out a visualization of this that makes sense, though I suspect one exists. Long and short, these decorators don't do anything.